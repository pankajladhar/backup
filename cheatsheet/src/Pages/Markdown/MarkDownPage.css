.MarkDownPage {
    padding-left: 260px;
}

.MarkDownPage__Content {
    max-width: 1000px;
    padding: 24px 32px 16px;
    margin: 0 auto;
    display: block;
}

.MarkDownPage__Sidebar {
    background-color: #f5f5f5;
    border: #e8e8e8 solid;
    border-width: 0 1px 0 0;
    position: fixed;
    top: 0;
    left: 0;
    bottom: 0;
    width: 260px;
    overflow: auto;
}

.MarkDownPage__Header {
    display: none;
}

@media (max-width: 767px) { 
    .MarkDownPage { 
        padding-left:  0;
        display: flex;
        flex-direction: column;
    }

    .MarkDownPage__Content {
        padding: 74px 16px 16px;
    }

    .MarkDownPage__Header { 
        display: flex;
        position: fixed;
        width: 100%;
        justify-content: center;
        border-bottom: 1px #811c19 solid;
        padding-bottom: 10px;
        background-color: #fff;
    }

    .MarkDownPage__Header .Logo__Main {
        width: 200px;
        height: 30px;
    }

    .MarkDownPage__Sidebar {
        left: -260px;
    }

}