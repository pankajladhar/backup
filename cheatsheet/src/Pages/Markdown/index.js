import React, { Component } from 'react';
import SideBar from './../../Components/SideBar';
import { MarkDown, MDExample, MDExampleCode, MDExamplePreview } from './../../Components/MarkDown';
import markdownData from './../../Data/markdown';
import Logo from './../../Components/Logo';
import './MarkDownPage.css';

class Markdown extends Component {
    renderSideBar() {
        const options = markdownData.map((item) => {
            return {
                id: item.id,
                title: item.title
            }
        })

        return (
            <div className="MarkDownPage__Sidebar">
                <SideBar options={options} heading="Markdown" />
            </div>
        )
    }

    renderMarkDownComponent() {
        return (

            markdownData.map((item, index) => {
                const { title, description, id } = item;
                return (
                    <MarkDown id={id}
                        key={`${id}_${index}`}
                        title={title}
                        description={description}>
                        {
                            item.content.map((subItem, index) => {
                                const { code, preview, subTitle, subDescription } = subItem
                                return (
                                    <MDExample key={index} title={subTitle} description={subDescription}>
                                        <MDExampleCode>
                                            <div dangerouslySetInnerHTML={{ __html: code }} />
                                        </MDExampleCode>
                                        <MDExamplePreview>
                                            <div dangerouslySetInnerHTML={{ __html: preview }} />
                                        </MDExamplePreview>
                                    </MDExample>
                                )
                            })
                        }
                    </MarkDown>
                )
            })
        )
    }
    render() {
        return (
            <div className="MarkDownPage">
                <header className="MarkDownPage__Header">
                    <Logo />
                    <button className={`Navigation__ToggleBtn ${this.state.isMainNavActive ? 'Navigation__ToggleBtn__Active' : ''}`}
                        onClick={this.toggleBtnClick}>
                        <span className="Navigation__ToggleBtn__Line"></span>
                        <span className="Navigation__ToggleBtn__Line"></span>
                        <span className="Navigation__ToggleBtn__Line"></span>
                    </button>
                </header>
                {this.renderSideBar()}
                <main className="MarkDownPage__Content">
                    {this.renderMarkDownComponent()}
                </main>
            </div>
        );
    }
}

export default Markdown;