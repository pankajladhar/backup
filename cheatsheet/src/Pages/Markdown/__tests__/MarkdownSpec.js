import React from 'react';
import renderer from 'react-test-renderer';
import Markdown from './../index';
jest.mock('./../../../Components/MarkDown', () => ({
    MarkDown: 'MarkDown',
    MDExample: 'MDExample',
    MDExampleCode: 'MDExampleCode',
    MDExamplePreview: 'MDExamplePreview'
}))

jest.mock('./../../../Components/SideBar', () => 'SideBar')
jest.mock('./../../../Data/markdown', () => (
    [
        {
            id: "headings",
            title: "Headings",
            description: "description",
            content: [{
                code: "code",
                preview: "preview",
                subTitle: "subTitle",
                subDescription: 'subDescription'
            }]
        }
    ]
))

describe('Markdown', () => {
    it('renders correctly', () => {
        let tree = renderer.create(
            <Markdown />
        );
        expect(tree.toJSON()).toMatchSnapshot();
    });
});