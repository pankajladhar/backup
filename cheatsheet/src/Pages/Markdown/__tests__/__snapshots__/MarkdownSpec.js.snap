// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown renders correctly 1`] = `
<div
  className="MarkDownPage"
>
  <div
    className="MarkDownPage__Sidebar"
  >
    <SideBar
      heading="Markdown"
      options={
        Array [
          Object {
            "id": "headings",
            "title": "Headings",
          },
        ]
      }
    />
  </div>
  <main
    className="MarkDownPage__Content"
  >
    <MarkDown
      description="description"
      id="headings"
      title="Headings"
    >
      <MDExample
        description="subDescription"
        title="subTitle"
      >
        <MDExampleCode>
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "code",
              }
            }
          />
        </MDExampleCode>
        <MDExamplePreview>
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "preview",
              }
            }
          />
        </MDExamplePreview>
      </MDExample>
    </MarkDown>
  </main>
</div>
`;
